docs/FAQ.md

```diff
@@ -0,0 +1,19 @@
+# FAQ for s2n-tls
+
+### Why is the ClientHello version reported as TLS 1.2 but the handshake itself negotiated TLS 1.3?
+The TLS 1.3 spec uses the TLS 1.2 version in ClientHello and on record layers for compatibility with middleboxes and poor server implementations. There is a separate ClientHello extension which lists the actual versions supported.
```

Comment:
You might want to add the "Why does the record say TLS1.1" (: We set it to TLS1.1 on the ClientHello for compatibility reasons:
> legacy_record_version:  MUST be set to 0x0303 for all records
      generated by a TLS 1.3 implementation other than an initial
      ClientHello (i.e., one not generated after a HelloRetryRequest),
      where it MAY also be 0x0301 for compatibility purposes.  This
      field is deprecated and MUST be ignored for all purposes.
      Previous versions of TLS would use other values in this field
      under some circumstances.

Maybe "Why is the ClientHello version reported as TLS 1.1 / 1.2"?